app/src/main/java/com/android/unio/ui/home/Home.kt

```diff
@@ -102,152 +90,199 @@ fun HomeScreen(
+    eventViewModel: EventViewModel,
+    paddingValues: PaddingValues
+) {
 
-                  // Underline to indicate selected tab with smooth sliding animation
-                  Box(modifier = Modifier.fillMaxWidth().padding(top = 4.dp)) {
-                    val selectedTabWidth =
-                        if (selectedTab == "All") allTabWidth else followingTabWidth
-                    Box(
-                        modifier =
-                            Modifier.offset(
-                                    x =
-                                        ((animatablePosition.value *
-                                                (followingTabXCoordinate - allTabXCoordinate) +
-                                                allTabXCoordinate) / density)
-                                            .dp - horizontalHeaderPadding)
-                                .width(selectedTabWidth)
-                                .height(2.dp)
-                                .background(Color.Blue)
-                                .testTag(HomeTestTags.UNDERLYING_BAR))
-                  }
-                }
-              }
+  val context = LocalContext.current
+  val events by eventViewModel.events.collectAsState()
+  // Event List
+  if (searchQuery.isNotEmpty() &&
+      (searchState == SearchViewModel.Status.SUCCESS ||
+          searchState == SearchViewModel.Status.LOADING)) {
+    if (searchResults.isEmpty()) {
+      Box(
+          modifier = Modifier.fillMaxSize().padding(paddingValues),
+          contentAlignment = Alignment.Center) {
+            Text(context.getString(R.string.explore_search_no_results))
+          }
+    } else {
+      EventList(navigationAction, searchResults, userViewModel, eventViewModel)
+    }
+  } else if (events.isNotEmpty()) {
+    EventList(navigationAction, events, userViewModel, eventViewModel)
+  } else {
+    Box(
+        modifier = Modifier.fillMaxSize().padding(paddingValues),
+        contentAlignment = Alignment.Center) {
+          Text(
+              modifier = Modifier.testTag(HomeTestTags.EMPTY_EVENT_PROMPT),
+              text = context.getString(R.string.event_no_events_available))
+        }
+  }
+}
+
+@Composable
+fun EventList(
+    navigationAction: NavigationAction,
+    events: List<Event>,
+    userViewModel: UserViewModel,
+    eventViewModel: EventViewModel
+) {
+  LazyColumn(
+      contentPadding = PaddingValues(vertical = 8.dp),
+      modifier = Modifier.fillMaxSize().padding(horizontal = 32.dp)) {
+        items(events) { event -> EventCard(navigationAction, event, userViewModel, eventViewModel) }
+      }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TopBar(
+    navigationAction: NavigationAction,
+    searchState: SearchViewModel.Status,
+    searchResults: List<Event>,
+    userViewModel: UserViewModel,
+    searchViewModel: SearchViewModel,
+    eventViewModel: EventViewModel,
+    paddingValues: PaddingValues
+) {
+  val nbOfTabs = 2
```

Comment:
Maybe extract this constant ? I'm being picky.

Reply:
I don't think the constant being here is bad practice, although I am not 100% sure of it. I'll leave it for now and we can see later what we would do with this kind of constants